sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java

```diff
@@ -110,14 +110,16 @@ protected void append(@NotNull ILoggingEvent eventObject) {
     event.setLogger(loggingEvent.getLoggerName());
     event.setLevel(formatLevel(loggingEvent.getLevel()));
 
-    final ThrowableProxy throwableInformation = (ThrowableProxy) loggingEvent.getThrowableProxy();
-    if (throwableInformation != null) {
-      final Mechanism mechanism = new Mechanism();
-      mechanism.setType(MECHANISM_TYPE);
-      final Throwable mechanismException =
-          new ExceptionMechanismException(
-              mechanism, throwableInformation.getThrowable(), Thread.currentThread());
-      event.setThrowable(mechanismException);
+    if (loggingEvent.getThrowableProxy() instanceof ThrowableProxy) {
```

Comment:
There's nothing else we can do here. In case we can't cast to `ThrowableProxy`, we cannot access the `Throwable` meaning we can't report the exception to Sentry.